gravity-forms/gw-time-sensitive-choices.php

```diff
@@ -39,64 +42,175 @@ public function __construct( $args = array() ) {
+						}
+					}
+
+					self.init = function() {
+
+						self.$target = $( '#input_{0}_{1}'.format( self.formId, self.fieldId ) );
+
+						self.evaluateChoices();
+
+						if ( self.dateFieldId ) {
+							self.$date = $( '#input_{0}_{1}'.format( self.formId, self.dateFieldId ) );
+							self.$date.on( 'change', function() {
+								var selectedDate = self.$date.datepicker( 'getDate' );
+								var currentDate  = self.getCurrentServerTime();
+								// Is future date?
+								if ( selectedDate > currentDate && selectedDate.getDate() > currentDate.getDate() ) {
+									self.enableChoices();
+								} else if ( selectedDate < currentDate && selectedDate.getDate() < currentDate.getDate() ) {
+									self.disableChoices();
+								} else {
+									self.evaluateChoices();
+								}
+
+							} );
+						}
+
+					};
+
+					self.evaluateChoices = function( mode ) {
+
+						var isDisabled;
+						var currentTime = self.getCurrentServerTime();
+
+						self.$target.find( 'option' ).each( function() {
+							switch ( mode ) {
+								case 'enable':
+									isDisabled = false;
+									break;
+								case 'disable':
+									isDisabled = true;
+									break;
+								default:
+									isDisabled = this.value && self.getChoiceTime( this.value ) < currentTime;
+							}
+							// This addresses placeholders specifically... not sure what other exceptions we'll need to make.
+							if ( this.value == '' ) {
+								isDisabled = false;
+							}
+							$( this ).prop( 'disabled', isDisabled );
+						} );
+
+					}
+
+					self.enableChoices = function() {
+						self.evaluateChoices( 'enable' );
+					}
+
+					self.disableChoices = function() {
+						self.evaluateChoices( 'disable' );
+					}
+
+					self.getChoiceTime = function( choiceTime ) {
+						var date = self.parseTime( choiceTime );
+						date.setMinutes( date.getMinutes() - self.buffer );
+						return date;
+					}
+
+					/**
+					 * @see https://stackoverflow.com/a/338439/227711
+					 * @param timeString
+					 * @returns {null|Date}
+					 */
+					self.parseTime = function( timeString ) {
+
+						if ( timeString == '' ) {
+							return null;
+						}
+
+						var date = new Date();
+						var time = timeString.match( /(\d+)(:(\d\d))?\s*(p?)/i );
+
+						date.setHours( parseInt( time[1], 10 ) + ( ( parseInt( time[1], 10 ) < 12 && time[4] ) ? 12 : 0 ) );
+						date.setMinutes( parseInt( time[3], 10 ) || 0 );
+						date.setSeconds( 0, 0 );
+
+						return date;
+					}
+
+					self.getCurrentServerTime = function() {
+						var date = new Date();
+						return self.convertTimezone( date, self.serverTimezone );
+					}
+
+					/**
+					 * @see https://stackoverflow.com/a/338439/227711
+					 * @param date
+					 * @param tzString
+					 * @returns {Date}
+					 */
+					self.convertTimezone = function( date, tzString ) {
+						return new Date( ( typeof date === 'string' ? new Date( date ) : date ).toLocaleString( 'en-US', { timeZone: tzString } ) );
```

Comment:
This doesn't work if the WP Timezone is set to a UTC timezone.

![](https://gwiz.io/3FlQsYm)

Produces this error:

```
Uncaught RangeError: Invalid time zone specified: 
    at Date.toLocaleString (<anonymous>)
    at window.GWTimeSensitiveChoices.self.convertTimezone (?gf_page=preview&id=6&unrequire=1:227)
    at window.GWTimeSensitiveChoices.self.getCurrentServerTime (?gf_page=preview&id=6&unrequire=1:217)
    at window.GWTimeSensitiveChoices.self.evaluateChoices (?gf_page=preview&id=6&unrequire=1:158)
    at window.GWTimeSensitiveChoices.self.init (?gf_page=preview&id=6&unrequire=1:134)
    at new window.GWTimeSensitiveChoices (?gf_page=preview&id=6&unrequire=1:230)
    at HTMLDocument.<anonymous> (?gf_page=preview&id=6&unrequire=1:239)
    at HTMLDocument.dispatch (jquery.min.js?ver=3.6.0:2)
    at HTMLDocument.v.handle (jquery.min.js?ver=3.6.0:2)
    at Object.trigger (jquery.min.js?ver=3.6.0:2)
```

Reply:
Oof. Good catch. Intended to revisit this today but will have to wait until Friday.

Reply:
@scyt This is fixed and will now work with both types of configured timezones. Could you retest?

Reply:
Works in my tests.